Lib/test/test_concurrent_futures.py

```diff
@@ -114,8 +116,13 @@ def setUp(self):
         except NotImplementedError as e:
             self.skipTest(str(e))
         self._prime_executor()
+        self.timer = threading.Timer(self.timeout, self._fail_on_deadlock)
+        self.timer.start()
 
     def tearDown(self):
+        self.timer.cancel()
+        self.timer.join()
+        del self.timer
```

Comment:
This is not really necessary I think.

Reply:
The `del` is necessary to make sure the thread is `gc`-ed before the end of the test method.
Without it, I get some `stopped` dangling threads with the python test suite.

Reply:
Calling `del` does not guarantee that the deleted object will be instantaneously garbage collected though.

> Without it, I get some stopped dangling threads with the python test suite.

This is reported as a warning by the test runner? If so, maybe add a comment to explain that the call to `del` is necessary to avoid the dangling threads detector of the test runner to issue a warning.